grazer/src/main/java/edu/asu/diging/grazer/core/fileupload/service/impl/FileUploadServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package edu.asu.diging.grazer.core.fileupload.service.impl;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.security.Principal;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import edu.asu.diging.grazer.core.domain.ITransformationFilesMetadata;
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesImpl;
+import edu.asu.diging.grazer.core.domain.impl.TransformationFilesMetadataImpl;
+import edu.asu.diging.grazer.core.fileupload.db.impl.FileMetadataDatabaseConnection;
+import edu.asu.diging.grazer.core.fileupload.service.IFileUploadService;
+import edu.asu.diging.grazer.web.fileUpload.FileUploadFormImpl;
+
+@Service
+@PropertySource(value = "classpath:config.properties")
+public class FileUploadServiceImpl implements IFileUploadService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private FileMetadataDatabaseConnection connection;
+
+    @Autowired
+    private Environment env;
+    
+    public synchronized String createID() {
+        return UUID.randomUUID().toString();
+    } 
+    
+    String createDirectory() {
+        File directory = new File(env.getProperty("transformation.file.dir") + File.separator + createID());
+        while(directory.exists()) {
+            directory = new File(env.getProperty("transformation.file.dir") + File.separator + createID());
+        }
+        directory.mkdir();
+        return directory.toString();
+    }
+    
+    void uploadFiles(CommonsMultipartFile[] multipartFiles) throws IOException {
+
+        String directory = createDirectory();
+
+        for(CommonsMultipartFile file: multipartFiles) {
+            try {      
+                String serverFileName = directory + File.separator + file.getOriginalFilename();
+                File serverFile = new File(serverFileName);
+
+                BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(serverFile));
+                stream.write(file.getBytes());
+                stream.close();
+            } catch(IOException e) {
+                logger.error("You failed to upload " + file.getOriginalFilename(), e);
+                throw new IOException(e);
+            } 
+        } 
+    }
```

Comment:
how do you know later on which files belongs to a transformation? It seems like the path to a file is never stored with the metadata.